arangod/Cluster/ResignShardLeadership.cpp

```diff
@@ -130,15 +119,31 @@ bool ResignShardLeadership::first() {
     // for now but we will not accept any replication operation from any
     // leader, until we have negotiated a deal with it. Then the actual
     // name of the leader will be set.
-    col->followers()->setTheLeader(LeaderNotYetKnownString);  // resign
-    res = methods->abort();                                   // unlock
+    for (auto const& s : shards) {
+      auto col = vocbase->lookupCollection(s);
+      if (col == nullptr) {
+        std::stringstream error;
+        error << "Failed to lookup local collection " << collection
+              << " in database " << database;
+        LOG_TOPIC("e06ca", ERR, Logger::MAINTENANCE)
+            << "ResignLeadership: " << error.str();
+        result(TRI_ERROR_ARANGO_DATA_SOURCE_NOT_FOUND, error.str());
+        return false;
+      }
+
+      col->followers()->setTheLeader(LeaderNotYetKnownString);  // resign
+    }
+
+    res = methods->abort();  // unlock
     if (res.fail()) {
       LOG_TOPIC("10c35", ERR, Logger::MAINTENANCE)
           << "Failed to abort transaction during resign leadership: " << res;
     }
 
-    transaction::cluster::abortLeaderTransactionsOnShard(col->id());
-
+    for (auto const& s : shards) {
+      auto col = vocbase->lookupCollection(s);
+      transaction::cluster::abortLeaderTransactionsOnShard(col->id());
```

Comment:
nit: it is theoretically possible that the lookup call _now_ produces a nullptr, because the shard was dropped on this server after the exclusive lock was released.
This may be an edge case, but I think it is better to check here if `col` is a nullptr or not.